main.js

```diff
@@ -0,0 +1,63 @@
+function delay(ms) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+
+async function fetchUsers(includeHeader) {
+  const statusElement = document.getElementById("status");
+  const elements = document.getElementById("user-list");
+
+  statusElement.textContent = "Loading...";
+  elements.textContent = "";
+
+  console.log("Fetching users...");
+
+  try {
+    const headers = includeHeader ? { "x-api-key": "reqres-free-v1" } : {};
+
+    const response = await fetch("https://reqres.in/api/users?delay=1", {
+      headers,
+    });
+
+    if (!response.ok) {
+      throw new Error("Network response was not ok");
+    }
+
+    const data = await response.json();
+
+    await delay(1000);
+
+    if (!data || !data.data || data.data.length === 0) {
```

Comment:
This will cover all 3 cases
```suggestion
    if (!data?.data?.length) {
```